### ISSUE TITLE
the way of loading the weight

### ISSUE THREAD

**Issue by wenouyang** [nicolov/segmentation_keras#8]
Hi nicolov,

In the post, you explained how to do the weight conversion. Due to the development environment constraints, it is a little bit hard for me to follow exactly your step. 

In [keras blog](https://blog.keras.io/category/demo.html), author also show a way to load VGG16 weight from Keras directly. Do you think this weight can be used for your implementation? Do we have to use the converted caffe model weight for pascal_voc. The data set I will be using is of different domain with the data set published in the paper. Thanks for your advice.

![capture](https://cloud.githubusercontent.com/assets/5430158/22951672/a6386214-f2cf-11e6-88c5-52afaea029d7.JPG)

**Comment by nicolov** [comment]
I've seen that, but what often happens is that these pretrained models have
subtle differences (e.g. normalization, or layer naming) that create issues
down the line. Why don't you just spin up a new VM to do the conversion? It
won't need much computing power.

**Comment by wenouyang** [comment]
Hi nicolov, 

Thanks for the reply.

By the way, I am working on some semantic segmentation work, which has data set sometimes of this kind of scale: 300~400 images with 1024*1024 as the size. In the masked training images, about 5%~10% areas to be labelled as foreground (in other words, we are dealing with two class segmentation). 

Currently, I generate a lot of training sets by randomly sampling the original images. The training set is of 128 * 128 size.  In other words, the trained model is based on 128 * 128 patches. In the predicted stage, I just stitch the 128 * 128 predicted masks together to generate the whole image mask. Is this the right approach for leveraging your framework for semantic segmentation? Thanks a lot.

**Comment by nicolov** [comment]
That sounds about right, and it's very similar to the original
paper/implementation. I would also make sure to exclude patches without
significant foreground.

**Comment by wenouyang** [comment]
Hi nicolov,

Would you like to explain more on the reasoning of "make sure to exclude patches without significant foreground".

Will this cause a lot of false positive?

Thanks.

**Comment by nicolov** [comment]
As you mentioned already, you have many more background than foreground
pixels in your training set. To make up for this class imbalance, you can
bias the sampling by rejecting patches with no foreground pixels at all.
